File: src/my_project/easy_problems/from151To200/toeplitz_matrix.py
from typing import List
from typing import List 

class Solution:
    def isToeplitzMatrix(self, matrix: List[List[int]]) -> bool:
        answer: bool = True

        if len(matrix) ==1 or len(matrix[0]) ==1:
            return answer
        for i in range(1,len(matrix)):
            
            for j in range(1,len(matrix[0])):
                if matrix[i][j] != matrix[i-1][j-1]:
                    answer = False 
                    return answer 

        return answer


solution = Solution()

print(solution.isToeplitzMatrix(matrix = [[1,2,3,4],[5,1,2,3],[9,5,1,2]]))

print(solution.isToeplitzMatrix(matrix = [[1,2],[2,2]]))

print(solution.isToeplitzMatrix(matrix = [[18],[66]]))

print(solution.isToeplitzMatrix(matrix = [[11,74,0,93],[40,11,74,7]]))

File: my_project_typescript/easy_problems_typescript/from51To100/toeplitzMatrix.ts
function isToeplitzMatrix(matrix: number[][]): boolean {

    let answer: boolean = true 

    if (matrix.length === 1 || matrix[0].length == 1){
        return answer 
    }

    for (let i = 1; i < matrix.length; i++){
        for (let j = 1; j < matrix[0].length; j++){
            if (matrix[i][j] !== matrix[i-1][j-1]){
                answer = false 
                return false 
            }
        }
    }

    return answer 

};

console.log(isToeplitzMatrix([[1,2,3,4],[5,1,2,3],[9,5,1,2]]))

console.log(isToeplitzMatrix([[1,2],[2,2]]))

console.log(isToeplitzMatrix([[18],[66]]))

console.log(isToeplitzMatrix( [[11,74,0,93],[40,11,74,7]]))
